# HG changeset patch
# User Konstantinos Poulios <logari81@googlemail.com>
# Date 1289127096 -3600
#      Sun Nov 07 11:51:36 2010 +0100
# Node ID 836427db633bc21f1f6f178cf6153fe395dbf44d
# Parent  d17cb8a1271d00dafa044375d5f00d1797ab3e83
Make colorbar function aware of dataaspect settings

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-11-07  Konstantinos Poulios  <logari81@googlemail.com>
+
+	* plot/colorbar.m: Make colorbar function aware of dataaspect settings.
+
 2010-11-06  Marco Caliari <marco.caliari@univr.it>
 
 	* polynomial/spline.m: Fit a parabola for input with 3
 	elements (bug #31098).
 
 2010-11-04  Rik  <octave@nomad.inbox5.com>
 
 	* plot/__fltk_ginput__.m: Use semicolons to prevent internal
diff --git a/scripts/plot/colorbar.m b/scripts/plot/colorbar.m
--- a/scripts/plot/colorbar.m
+++ b/scripts/plot/colorbar.m
@@ -161,16 +161,18 @@ function h = colorbar (varargin)
                   "yliminclude", "off", "zliminclude", "off",
                   "deletefcn", {@deletecolorbar, cax, obj});
 
     set (cax, "deletefcn", {@resetaxis, obj});
 
     addlistener (ax, "clim", {@update_colorbar_clim, hi, vertical})
     addlistener (ax, "plotboxaspectratio", {@update_colorbar_axis, cax, obj})
     addlistener (ax, "plotboxaspectratiomode", {@update_colorbar_axis, cax, obj})
+    addlistener (ax, "dataaspectratio", {@update_colorbar_axis, cax, obj})
+    addlistener (ax, "dataaspectratiomode", {@update_colorbar_axis, cax, obj})
     addlistener (ax, "position", {@update_colorbar_axis, cax, obj})
 
   endif
 
   if (nargout > 0)
     h = cax;
   endif
 endfunction
@@ -255,17 +257,18 @@ function update_colorbar_axis (h, d, cax
 endfunction
 
 function [pos, cpos, vertical, mirr] = __position_colorbox__ (cbox, obj, cf)
 
   ## This will always represent the position prior to adding the colorbar.
   pos = obj.position;
   sz = pos(3:4);
 
-  if (strcmpi (obj.plotboxaspectratiomode, "manual"))
+  if (strcmpi (obj.plotboxaspectratiomode, "manual")
+      || strcmpi (obj.dataaspectratiomode, "manual"))
     if (isempty (strfind (cbox, "outside")))
       scale = 1.0;
     else
       scale = 0.8;
     endif
     if (isempty (strfind (cbox, "east")) && isempty (strfind (cbox, "west")))
       scale = [1, scale];
     else
@@ -329,17 +332,18 @@ function [pos, cpos, vertical, mirr] = _
       origin = pos(1:2) + [0.05, 0.05] .* sz + off;
       sz = sz .* [0.06, 1.0] .* 0.9;
       mirr = true;
       vertical = true;
   endswitch
 
   cpos = [origin, sz];
 
-  if (strcmpi (obj.plotboxaspectratiomode, "manual"))
+  if (strcmpi (obj.plotboxaspectratiomode, "manual")
+      || strcmpi (obj.dataaspectratiomode, "manual"))
     obj.position = pos;
     actual_pos = __actual_axis_position__ (obj);
     if (strfind (cbox, "outside"))
       scale = 1.0;
     else
       scale = 0.9;
     endif
     if (sz(1) > sz(2))
